### ISSUE TITLE
value-list: screen reader instructions are missing when drag handle is activated

### ISSUE THREAD

**Issue by anveshmekala** [Esri/calcite-components#6401]
### Actual Behavior

when the user tabs into the dragHandle and activate with space key, no instruction is displayed in screen reader. 

### Expected Behavior

screen reader instructions should be displayed when the dragHandle is activated/ deactivated. 

### Reproduction Sample

https://codepen.io/ANVESHMEKALA/pen/BaPqVgV?editors=1010

### Reproduction Steps

1. Start the screen reader mode in the codepen . 
2. Try navigation into the first value-list-item. Make sure the screen reader reads out  the instruction 
3. Hit space to activate the dragHandle.
4. Observe no instruction is announced by the screen reader.

### Reproduction Version

bet.99

### Relevant Info

cancelling out the event [here](https://github.com/Esri/calcite-components/pull/5709/files#diff-9cb5506aef9cc2ff9af5ee350a918a33151cda203737921476258b8014387de7R179) is not allowing the screen reader text to update.

### Esri team

N/A

**Comment by geospatialem** [comment]
This resolution will likely only apply to Mac and VoiceOver, as there was no previous support for Windows machines with NVDA and JAWS. cc https://github.com/Esri/calcite-components/issues/3982#issuecomment-1206487248. 

In the long term we should explore some alternatives with the `list` component when we add dragging functionality to the component.

**Comment by geospatialem** [comment]
Verified on `1.0.8-next.3` with Safari and VoiceOver. Both NVDA and JAWS are unable to access when activated, related to #3982.
